lib/rules/indent.js

```diff
@@ -0,0 +1,478 @@
+/*eslint no-use-before-define:[2, "nofunc"]*/
+/**
+ * @fileoverview This option sets a specific tab width for your code
+ * @author Dmitriy Shekhovtsov
+ * @author Mike Sherov <mikesherov>
+ * @copyright 2015 Dmitriy Shekhovtsov. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    // indentation defaults: 4 spaces
+    var indentChar = " ";
+    var indentSize = 4;
+    var options = {switchCase: false};
+
+    var lines = null;
+    var indentStack = [0];
+    var linesToCheck = null;
+    var breakIndents = null;
+
+    if (context.options.length) {
+        if (context.options[0] === "tab") {
+            indentChar = "\t";
+            indentSize = 1;
+        } else if (typeof context.options[0] === "number") {
+            indentSize = context.options[0];
+        }
+
+        if (context.options[1]) {
+            var opts = context.options[1];
+            options.switchCase = opts.switchCase === true;
+            options.withStatement = opts.withStatement === true;
```

Comment:
`WithStatement` should always be included in `blockParents` below. There's no reason to force people to turn it on.